packages/library-base/src/commonMain/kotlin/io/realm/kotlin/log/RealmLogger.kt

```diff
@@ -38,12 +39,23 @@ public interface RealmLogger {
     /**
      * Log an event.
      */
-    public fun log(level: LogLevel, throwable: Throwable?, message: String?, vararg args: Any?)
+    public fun log(category: LogCategory, level: LogLevel, throwable: Throwable?, message: String?, vararg args: Any?) {
+        log(level, throwable, message, *args)
+    }
+
+    /**
+     * Log an event.
+     */
+    @Deprecated("Use log(category: LogCategory, level: LogLevel, throwable: Throwable?, message: String?, vararg args: Any?)", ReplaceWith("log"))
+    public fun log(level: LogLevel, throwable: Throwable?, message: String?, vararg args: Any?) {
```

Comment:
I guess we cannot leave the category-variant without a default implementation, as that would be a breaking change if users need to implement that, but feels like there is something wrong here. This would just recurse between the two log methods if ever called. I think it would be safer to have a 
```
// Don't overwrite this, but overwrite the category-variant instead
TODO("log() shouldn't be called. Overwrite the category-variant instead")
```
or something like that, so people would be guided the right way faster.

Reply:
I wanted to avoid a breaking change but I introduced a cycle. I have update the logic.

Reply:
The new implementation looks even more confusing than the old. I don't think it guides users if they don't see any output from `fun log(level: LogLevel, message: String)`!?